pallets/parachain-staking/src/lib.rs

```diff
@@ -1490,6 +1468,16 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		pub fn set_candidate_bond_to_zero(acc: &T::AccountId) -> DispatchResultWithPostInfo {
+			ensure!(Self::is_candidate(&acc), Error::<T>::CandidateDNE);
+			let actual_weight =
+				T::WeightInfo::set_candidate_bond_to_zero(<CandidatePool<T>>::get().0.len() as u32);
```

Comment:
That's esp important since this ultimately gets run inside a loop, we don't want to decode the entire thing for every single collator during the migration

Reply:
It always amuses me when measuring something to know how expensive it is actually causes it to become significantly more expensive.

I would actually change this to just use `T::MaxCandidates::get()` because (1) that's way cheaper than decoding or even reading storage, and (2) it's going to match our on-chain cases anyway.

Reply:
That sounds reasonable. I have applied the same change to 2 other places.